apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/RatingQuestionForm.tsx

```diff
@@ -33,6 +35,7 @@ export default function RatingQuestionForm({
             name="headline"
             value={question.headline}
             onChange={(e) => updateQuestion(questionIdx, { headline: e.target.value })}
+            isInvalid={isFaulty && question.headline.trim() === ""}
```

Comment:
why is the `&& question.headline.trim() === ""` needed? isn't this something the validation function should handle?

Reply:
Responsibility for marking a question as faulty or invalid lies with `isFaulty` To identify the specific faulty field within that particular question, the following expression is employed: `isFaulty && question.headline.trim() === ""`.

It is important to note that if `isFaulty` is false, there is no need to proceed with further checks, as the question is deemed valid. However, in cases where the question is indeed faulty, this expression helps pinpoint which field is at fault.

Reply:
Got it! :-)
The issue I am seeing is that we now have the validation rules in many different places. We define in `Validation.ts` when a question is invalid but additionally define rules in every input field when this is invalid. Ideally we can combine these to globally manage the validation rules.

Reply:
Long-term I would prefer a more standardised field validation that works with e.g. react hook form ( https://react-hook-form.com/get-started#Applyvalidation ) which we also use in other places in the app.

The tricky part is that we have a form for every question and have the save/publish actions far on the outside of this code. But I think if we find a good solution there using react hook form would be a good approach to define validation rules and also work with error messages, etc.